File: CSES_new/Dynamic Programming/DiceCombinations.cpp
#include <iostream>
#include <vector>

int main()
{
	int n;
	std::cin >> n;
	std::vector<long long> dp(n+1,0);
	const long long MOD = 1e9+7;
	dp[0] = 1;
	for(int i = 1; i <= n; i++)
	{
		for(int j = 1; j <= 6; j++)
		{
			if(i-j >= 0)
			{
				dp[i] += dp[i-j];
				dp[i] %= MOD;
			}
		}
	}
	std::cout << dp[n] << "\n";
	return 0;
}

File: CSES_new/Dynamic Programming/DiceCombinations.py
n = int(input())
dp = [0]*(n+1)
mod = int(1e9+7)
dp[0] = 1
for i in range(n+1):
	for j in range(1,7):
		if i-j >= 0:
			dp[i] += dp[i-j]
			dp[i] %= mod
print(dp[n])
